src/python/zquantum/core/interfaces/backend.py

```diff
@@ -128,25 +152,25 @@ def get_wavefunction(self, circuit: Circuit, **kwargs) -> Wavefunction:
             pyquil.Wafefunction: wavefunction object.
 
         """
-
-        raise NotImplementedError
+        self.number_of_circuits_run += 1
+        self.number_of_jobs_run += 1
 
     @abstractmethod
     def get_exact_expectation_values(
-        self, circuit: Circuit, qubit_operator: QubitOperator, **kwargs
+        self, circuit: Circuit, operator: SymbolicOperator, **kwargs
     ) -> ExpectationValues:
         """
         Calculates the expectation values for given operator, based on the exact quantum state produced by circuit.
 
         Args:
             circuit (core.circuit.Circuit): quantum circuit to be executed.
-            qubit_operator(openfermion): Operator for which we calculate the expectation value.
+            operator(openfermion): Operator for which we calculate the expectation value.
 
         Returns:
             ExpectationValues: object representing expectation values for given operator.
         """
-
-        raise NotImplementedError
+        wavefunction = self.get_wavefunction(circuit)
```

Comment:
I'm not sure whether putting it like this actually gives us any benefits?
Some simulators might have special functions for expectation values (e.g. `qHipster`). 
It's not a lot of code for people to write it and having `raise NotImplementedError` is much clearer IMO.

Reply:
Even though it's not a _lot_ of code, I still think it's good to have a default method that people can inherit without re-implementing. If someone wants to implement a new method, they can always override the default one

Reply:
Fine.